Mathlib/LinearAlgebra/TensorProduct/RightExactness.lean

```diff
@@ -0,0 +1,329 @@
+    LinearMap.range g ≤ (LinearMap.range (lTensor Q g)).comap (TensorProduct.mk R Q P q) := by
+  rintro x ⟨n, rfl⟩
+  exact ⟨q ⊗ₜ[R] n, rfl⟩
+
+lemma le_comap_range_rTensor (q : Q) :
+    LinearMap.range g ≤ (LinearMap.range (rTensor Q g)).comap
+      ((TensorProduct.mk R P Q).flip q) := by
+  rintro x ⟨n, rfl⟩
+  exact ⟨n ⊗ₜ[R] q, rfl⟩
+
+variable (Q) {g}
+
+theorem rTensor.Surjective (hg : Function.Surjective g) :
+  Function.Surjective (rTensor Q g) := by
+  intro z
+  induction z using TensorProduct.induction_on with
+  | zero => use 0; rw [map_zero]
+  | tmul p q =>
+      obtain ⟨n, rfl⟩ := hg p
+      use n ⊗ₜ[R] q
+      simp only [rTensor_tmul]
+  | add x y hx hy =>
+      obtain ⟨x, rfl⟩ := hx
+      obtain ⟨y, rfl⟩ := hy
+      use x + y
+      rw [map_add]
+
+theorem lTensor.Surjective (hg : Function.Surjective g) :
+  Function.Surjective (lTensor Q g) := by
+  intro z
+  induction z using TensorProduct.induction_on with
+  | zero => use 0; rw [map_zero]
+  | tmul q p =>
+      obtain ⟨n, rfl⟩ := hg p
+      use q ⊗ₜ[R] n
+      simp only [lTensor_tmul]
+  | add x y hx hy =>
+      obtain ⟨x, rfl⟩ := hx
+      obtain ⟨y, rfl⟩ := hy
+      use x + y
+      rw [map_add]
+
+end Semiring
+
+variable {R : Type _} [CommRing R]
+variable {M N P P' : Type _}
+  [AddCommGroup M] [AddCommGroup N] [AddCommGroup P]
+  [Module R M] [Module R N] [Module R P]
+
+variable {f : M →ₗ[R] N} {g : N →ₗ[R] P}
+
+variable (Q : Type _) [AddCommGroup Q] [Module R Q]
+
+variable (hfg : Exact f g) (hg : Function.Surjective g)
+
+private noncomputable
+def rTensor.inverse_ofRightInverse {h : P → N} (hgh : Function.RightInverse h g) :
+  P ⊗[R] Q →ₗ[R] N ⊗[R] Q ⧸ (LinearMap.range (rTensor Q f)) := by
+  rw [exact_iff] at hfg
+  apply TensorProduct.lift
+  apply LinearMap.mk₂ R (fun p q ↦ Submodule.Quotient.mk (h p ⊗ₜ[R] q))
+  · intro p p' q
+    rw [← Submodule.Quotient.mk_add, Submodule.Quotient.eq]
+    rw [← TensorProduct.add_tmul, ← TensorProduct.sub_tmul]
+    apply le_comap_range_rTensor f
+    rw [← hfg, mem_ker, map_sub, map_add]
+    simp only [hgh _, sub_self]
+  · intro r p q
+    rw [← Submodule.Quotient.mk_smul, Submodule.Quotient.eq]
+    rw [TensorProduct.smul_tmul', ← TensorProduct.sub_tmul]
+    apply le_comap_range_rTensor f
+    rw [← hfg, mem_ker, map_sub, map_smul]
+    simp only [hgh _, sub_self]
+  · intro p q q'
+    rw [← Submodule.Quotient.mk_add, Submodule.Quotient.eq]
+    rw [TensorProduct.tmul_add]
+    rw [add_sub_add_comm]
+    simp only [← TensorProduct.sub_tmul]
+    apply Submodule.add_mem
+    all_goals {
+      apply le_comap_range_rTensor f
+      rw [← hfg, mem_ker, map_sub]
+      simp only [hgh _, sub_self] }
+  · intro r p q
+    simp only [TensorProduct.tmul_smul, Submodule.Quotient.mk_smul]
+
+private noncomputable
+def rTensor.inverse :
+  P ⊗[R] Q →ₗ[R] N ⊗[R] Q ⧸ (LinearMap.range (rTensor Q f)) :=
+  rTensor.inverse_ofRightInverse Q hfg
+    (Function.Surjective.hasRightInverse hg).choose_spec
```

Comment:
```suggestion
  rTensor.inverse_ofRightInverse Q hfg
    (Function.rightInverse_surjInv hg)
```
which requires a very minor tweak to a later proof

Reply:
Done!